File: Toy-Language-Interpreter_C#/Toy-Language-Interpreter/DataTypes/MyIDictionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_Interpreter.DataTypes
{
    interface MyIDictionary<K, V>
    {
        void Clone(MyIDictionary<K, V> toClone);
        Dictionary<K, V> GetDict();
        V Get(K key);
        V Put(K key, V value);
        bool Remove(K key);
        bool ContainsKey(K key);
        bool ContainsValue(V value);
        List<K> KeySet();
        List<V> Values();
        int Size();
        bool IsEmpty();
        string ToString();
    }
}

File: Toy-Language-Interpreter/src/DataTypes/MyIDictionary.java
package DataTypes;

import Exception.InvalidKeyMyDictionaryException;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;

/**
 * Created by vladc on 22.10.2016.
 */
public interface MyIDictionary<K, V> {
    void clone(MyIDictionary<K, V> toClone);
    Map<K, V> getDict();
    V get(Object key) throws InvalidKeyMyDictionaryException;
    V put(K key, V value);
    V remove(Object key) throws InvalidKeyMyDictionaryException;
    V replace(K key, V value) throws InvalidKeyMyDictionaryException;
    boolean containsKey(Object key);
    boolean containsValue(Object value);
    Set<K> keySet();
    Collection<V> values();
    int size();
    boolean isEmpty();
}
